project/code/keychain/blockchain.py

```diff
@@ -0,0 +1,86 @@
+import hashlib
+import multilib
+import numpy as np
+
+from keychain.util import hash
+
+
+
+class Block:
```

Comment:
Is this given? or shall we rather just specify the class with empty methods they should fill in? (in which case, we should document exactly what each method does)

Reply:
At the moment this is "given", but maybe this makes it a bit to easy.